pennylane/tape/tapes/cv_param_shift.py

```diff
@@ -464,6 +464,7 @@ def parameter_shift_var(self, idx, params, **options):
 
         # Get <A>, the expectation value of the tape with unshifted parameters.
         evA_tape = self.copy()
+        evA_tape.set_parameters(params)

```

Comment:
A bug discovered in the parameter shift rule.

Reply:
Oh!

Reply:
Could this be a problem anywhere else?

Reply:
Not a the moment (I went through the existing gradient logic looking).

The reason why it needs to be added:

* The `params` that have been passed to the gradient methods have been unwrapped to NumPy arrays by the interface

* However, the tape itself is unmodified and still contains tensors internally.

So when we copy the tape, we need to make sure that all internal parameters get replaced with the unwrapped versions, since we are inside the interface.